src/backend/metal/src/device.rs

```diff
@@ -262,9 +375,51 @@ impl PhysicalDevice {
                 argument_buffers: Self::supports_any(device, ARGUMENT_BUFFER_SUPPORT) && false, //TODO
                 shared_textures: !Self::is_mac(device),
                 format_depth24_stencil8: device.d24_s8_supported(),
-                format_depth32_stencil8: true, //TODO: crashing the Metal validation layer upon copying from buffer
+                format_depth32_stencil8_filter: Self::is_mac(device), 
+                format_depth32_stencil8_none: !Self::is_mac(device),
                 format_min_srgb_channels: if Self::is_mac(&*device) {4} else {1},
                 format_b5: !Self::is_mac(device),
+                format_bc: Self::is_mac(device),
+                format_eac_etc: !Self::is_mac(device),
+                format_astc: Self::supports_any(device, ASTC_PIXEL_FORMAT_FEATURES),
+                format_r8unorm_srgb_all: Self::supports_any(device, R8UNORM_SRGB_ALL),
+                format_r8unorm_srgb_no_write: !Self::supports_any(device, R8UNORM_SRGB_ALL) && !Self::is_mac(device),
```

Comment:
frankly, it does sound easier to just assign the flags right here instead of storing booleans only to transform them to flags later on. The image format flags are more expressive than a pair of `xxx_all` and `xxx_no_write` booleans.